src/seedsigner/hardware/microsd.py

```diff
@@ -0,0 +1,58 @@
+import time
+
+from seedsigner.models.singleton import Singleton
+from seedsigner.models.threads import BaseThread
+
+class MicroSD(Singleton, BaseThread):
+	
+	ui_handler = None
+	settings_handler = None
+	
+	@classmethod
+	def get_instance(cls):
+		# This is the only way to access the one and only instance
+		if cls._instance is None:
+			# Instantiate the one and only instance
+			microsd = cls.__new__(cls)
+			cls._instance = microsd
+			
+			# explicitly call BaseThread __init__ since multiple class inheritance
+			BaseThread.__init__(microsd)
+	
+		return cls._instance
+	
+	def start_detection(self):
+		self.start()
+	
+	def run(self):
+		import socket, os, time
+		
+		from seedsigner.controller import Controller
+		
+		# explicitly only microsd add/remove detection in seedsigner-os
+		if Controller.HOSTNAME == Controller.SEEDSIGNER_OS:
+			
+			if os.path.exists("/tmp/mdev_socket"):
+				os.remove("/tmp/mdev_socket")
+			
+			server = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+			server.bind("/tmp/mdev_socket")
+			while self.keep_running:
+				
+				server.listen(1)
+				conn, addr = server.accept()
```

Comment:
Are any of these calls blocking? If not, I'm wondering if we need a `time.sleep()` in this loop so it doesn't eat too many resources.

Reply:
For future reference, here is an example of seedsigner OS passing an add message to the socket.

https://github.com/SeedSigner/seedsigner-os/blob/b4c95a2b3a4e16b23cc7c50cda2f608287c2d400/rpi-seedsigner/board/raspberrypi/rootfs-overlay/etc/mdev/mdev.sh#L8